utilities/node_activator/lib/node_activator.ex

```diff
@@ -10,23 +10,25 @@ defmodule NodeActivator do
   @doc """
   Activates Node.
   """
-  @spec run(binary()) :: {:ok, node() | pid()} | {:error, term()}
-  def run(name) do
-    if Node.alive?() do
-      {:ok, Node.self()}
-    else
-      case :os.type() do
-        {:unix, _} -> launch_epmd(port: @epmd_port)
-        {:win32, _} -> launch_epmd(port: @epmd_port, daemon: false)
-      end
+  @spec run(binary()) :: {:ok, node()}
+  def run(node_name_prefix) do
+    unless Node.alive?() do
+      empd_options =
+        case :os.type() do
+          {:unix, _} -> [port: @epmd_port]
+          {:win32, _} -> [port: @epmd_port, daemon: false]
+        end
 
+      launch_epmd(empd_options)
       Logger.info("wait launching epmd...")
       wait_launching_epmd(5)
       Logger.info("done.")
-      name = name |> name_with_random_key() |> String.to_atom()
+      name = generate_node_name(node_name_prefix)
       Logger.info("Node.start(#{name})")
       Node.start(name)
```

Comment:
Your code does not seem to work well when `Node.start/3` may return `{:error, term()}`.